From c889ef8d7ee9fe4e126d3e38143d0a2bf4908dba Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Sun, 20 Nov 2022 20:38:11 +0800
Subject: [PATCH] policy/modules/admin/ostree: fix ostree upgrade

Fixes:
avc:  denied  { create } for  pid=681 comm="ostree"
scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023 tclass=unix_dgram_socket
permissive=1

avc:  denied  { getattr } for  pid=705 comm="bwrap"
path="cgroup:[4026531835]" dev="nsfs" ino=4026531835
scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=system_u:object_r:nsfs_t:s0 tclass=file permissive=1

avc:  denied  { mount } for  pid=706 comm="bwrap" name="/" dev="tmpfs"
ino=1 scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=system_u:object_r:tmpfs_t:s0 tclass=filesystem permissive=1

avc:  denied  { remount } for  pid=706 comm="bwrap"
scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=system_u:object_r:device_t:s0 tclass=filesystem permissive=1

avc:  denied  { mount } for  pid=706 comm="bwrap" name="/" dev="devpts"
ino=1 scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=system_u:object_r:devpts_t:s0 tclass=filesystem permissive=1

avc:  denied  { mount } for  pid=706 comm="bwrap" name="/" dev="proc"
ino=1 scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=system_u:object_r:proc_t:s0 tclass=filesystem permissive=1

avc:  denied  { remount } for  pid=706 comm="bwrap"
scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=system_u:object_r:proc_t:s0 tclass=filesystem permissive=1

avc:  denied  { remount } for  pid=706 comm="bwrap"
scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=system_u:object_r:sysfs_t:s0 tclass=filesystem permissive=1

avc:  denied  { remount } for  pid=706 comm="bwrap"
scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=system_u:object_r:fs_t:s0 tclass=filesystem permissive=1

avc:  denied  { unmount } for  pid=706 comm="bwrap"
scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=system_u:object_r:fs_t:s0 tclass=filesystem permissive=1

avc:  denied  { unmount } for  pid=706 comm="bwrap"
scontext=root:sysadm_r:ostree_t:s0-s15:c0.c1023
tcontext=system_u:object_r:tmpfs_t:s0 tclass=filesystem permissive=1

Upstream-Status: Inappropriate [ostree specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/admin/ostree.te | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/policy/modules/admin/ostree.te b/policy/modules/admin/ostree.te
index 573ff71de..bc47824c8 100644
--- a/policy/modules/admin/ostree.te
+++ b/policy/modules/admin/ostree.te
@@ -31,6 +31,7 @@ allow ostree_t self:tcp_socket { accept listen };
 allow ostree_t self:msg { send receive };
 allow ostree_t self:file rw_file_perms;
 allow ostree_t self:netlink_kobject_uevent_socket create_socket_perms;
+allow ostree_t self:unix_dgram_socket create_socket_perms;
 
 kernel_read_crypto_sysctls(ostree_t)
 kernel_read_network_state(ostree_t)
@@ -79,8 +80,11 @@ files_map_non_auth_files(ostree_t)
 fs_getattr_all_dirs(ostree_t)
 fs_getattr_all_fs(ostree_t)
 
-fs_remount_dos_fs(ostree_t)
-fs_mount_xattr_fs(ostree_t)
+fs_read_nsfs_files(ostree_t)
+
+fs_mount_all_fs(ostree_t)
+fs_remount_all_fs(ostree_t)
+fs_unmount_all_fs(ostree_t)
 files_mounton_all_mountpoints(ostree_t)
 
 sysnet_dns_name_resolve(ostree_t);
-- 
2.25.1

